### PR TITLE
Add FXIOS-6029 [v114] #13679 Enable homepanel in browser coordinator

### PR DESCRIPTION
**Pull request by razvanlitianu**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6029)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/13679)

### Description
Handle homepanel routes in browser coordinator.

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

**Review by OrlaM — APPROVED** [review]

**Review by lmarceau — COMMENTED** [review]
Some comments related to tests, and 1 nit for spacing

File: Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift [hunk 1]
```diff
@@ -231,4 +254,148 @@ final class BrowserCoordinatorTests: XCTestCase {
         trackForMemoryLeaks(subject, file: file, line: line)
         return subject
     }
+
+    func testHandleHomepanelBookmarks_returnsTrue() {
```
[the remark this answers is not in the document]
Comment (razvanlitianu) [reply Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift:258]: They start with word `test` so I'm not sure how someone could confuse them with helpers. I can change it.

File: Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift [hunk 2]
```diff
@@ -152,73 +155,93 @@ final class BrowserCoordinatorTests: XCTestCase {
...
+            expectation.fulfill()
         }
         mbvc.handleQueryCalled = { queryCalled in
             XCTAssertEqual(query, queryCalled)
+            expectation.fulfill()
         }
         subject.browserViewController = mbvc
         let result = subject.handle(route: .searchQuery(query: query))
         XCTAssertTrue(result)
+        wait(for: [expectation], timeout: 1)
```
[the remark this answers is not in the document]
Comment (razvanlitianu) [reply Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift:170]: Adding properties is also acceptable, but think about all the routes we will add and how many properties we need to hold for each and the states that grow with it. Having closures makes a direct connection between the function that's called and the parameters passed that we can assert.

**Review by tharkhay — APPROVED** [review]

File: Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (tharkhay) [reply Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift:223]: +223 to +223

File: Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (tharkhay) [reply Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift]: []()

**Comment by tharkhay** [comment]
Client/Coordinators/Browser/BrowserCoordinator.swift

File: Client/Coordinators/Browser/BrowserCoordinator.swift [hunk 3]
```diff
@@ -149,6 +149,24 @@ class BrowserCoordinator: BaseCoordinator, LaunchCoordinatorDelegate, BrowserDel
             return false
         }
     }
+    private func handle(homepanelSection section: Route.HomepanelSection) {
```
[the remark this answers is not in the document]
Comment (tharkhay) [reply Client/Coordinators/Browser/BrowserCoordinator.swift]: Client/Coordinators/Browser/BrowserCoordinator.swift

File: Tests/ClientTests/Mocks/MockBrowserViewController.swift [hunk 4]
```diff
@@ -6,24 +6,72 @@ import UIKit
...
+    var switchToPrivacyModeCount = 0
+    var switchToTabForURLOrOpenCount = 0
+    var openBlankNewTabCount = 0
+    var handleQueryCount = 0
+    var showLibraryCount = 0
+    var openURLInNewTabCount = 0
 
     override func switchToPrivacyMode(isPrivate: Bool) {
-        switchToPrivacyModeCalled(true)
+        switchToPrivacyModeCalled = true
```
Comment (lmarceau) [inline Tests/ClientTests/Mocks/MockBrowserViewController.swift:38]: nit, the called property is accounted for when the count is 1 or more. We can chat over this and merge this PR in the meantime

**Review by lmarceau — APPROVED** [review]

**Pull request merged**